src/Umbraco.Web/Routing/ContentFinderByNiceUrlAndTemplate.cs

```diff
@@ -1,57 +1,58 @@
 using Umbraco.Core.Logging;
 using Umbraco.Core.Models;
 using Umbraco.Core;
+using Umbraco.Core.Configuration;
 
 namespace Umbraco.Web.Routing
 {
-	/// <summary>
-	/// Provides an implementation of <see cref="IContentFinder"/> that handles page nice urls and a template.
-	/// </summary>
-	/// <remarks>
-	/// <para>Handles <c>/foo/bar/template</c> where <c>/foo/bar</c> is the nice url of a document, and <c>template</c> a template alias.</para>
-	/// <para>If successful, then the template of the document request is also assigned.</para>
-	/// </remarks>
+    /// <summary>
+    /// Provides an implementation of <see cref="IContentFinder"/> that handles page nice urls and a template.
+    /// </summary>
+    /// <remarks>
+    /// <para>Handles <c>/foo/bar/template</c> where <c>/foo/bar</c> is the nice url of a document, and <c>template</c> a template alias.</para>
+    /// <para>If successful, then the template of the document request is also assigned.</para>
+    /// </remarks>
     public class ContentFinderByNiceUrlAndTemplate : ContentFinderByNiceUrl

```

Comment:
What are you expecting here? If I use URL /foo/bar/template... Should we render /foo/bar with the original template, or just 404? I think we should 404. And then all we need is to return null immediately in this finder.